In [15]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline 
pd.set_option('display.max_rows',1000)
pd.set_option('display.max_columns', 2000)

In [16]:
worlds_data_csv = 'C:/Users/javier/Desktop/projects/League_of_Legends_data_players_team_projects/datasets/2017-worlds-match-data-OraclesElixir-2017-10-15.csv'
worlds_data_xls = 'C:/Users/javier/Desktop/projects/League_of_Legends_data_players_team_projects/datasets/2017-worlds-match-data-OraclesElixir-2017-10-15.xlsx'

In [32]:
#Understanding Data
play_in_df = pd.read_excel(worlds_data)

In [33]:
play_in_df.head(100)


gameid                                                url league  \
0   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
1   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
2   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
3   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
4   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
5   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
6   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
7   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
8   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
9   1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
10  1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
11  1002360213  http://matchhistory.na.leagueoflegends.com/en/...     WC   
12  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
13  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
14  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
15  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
16  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
17  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
18  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
19  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
20  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
21  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
22  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
23  1002360222  http://matchhistory.na.leagueoflegends.com/en/...     WC   
24  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
25  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
26  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
27  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
28  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
29  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
30  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
31  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
32  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
33  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
34  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
35  1002360229  http://matchhistory.na.leagueoflegends.com/en/...     WC   
36  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
37  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
38  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
39  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
40  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
41  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
42  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
43  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
44  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
45  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
46  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
47  1002360235  http://matchhistory.na.leagueoflegends.com/en/...     WC   
48  1002360244  http://matchhistory.na.leagueoflegends.com/en/...     WC   
49  1002360244  http://matchhistory.na.leagueoflegends.com/en/...     WC   
50  1002360244  http://matchhistory.na.leagueoflegends.com/en/...     WC   
51  1002360244  http://matchhistory.na.leagueoflegends.c

In [34]:
play_in_df.describe()

gameid          date         week       patchno     playerid  \
count  1.080000e+03   1080.000000  1080.000000  1.080000e+03  1080.000000   
mean   1.002374e+09  43012.076615     2.158889  7.180000e+00    29.583333   
std    1.270939e+04      7.717896     0.809063  1.004107e-13    57.677176   
min    1.002360e+09  43001.001840     1.100000  7.180000e+00     1.000000   
25%    1.002360e+09  43004.111146     1.400000  7.180000e+00     3.750000   
50%    1.002380e+09  43013.682193     2.150000  7.180000e+00     6.500000   
75%    1.002390e+09  43020.124306     3.100000  7.180000e+00     9.250000   
max    1.002391e+09  43023.150648     3.400000  7.180000e+00   200.000000   

        gamelength       result            k            d            a  \
count  1080.000000  1080.000000  1080.000000  1080.000000  1080.000000   
mean     34.814815     0.500000     3.270370     3.274074     8.509259   
std       9.122293     0.500232     4.312734     4.080841    10.862265   
min      20.866667     0.000000     0.000000     0.000000     0.000000   
25%      26.916667     0.000000     0.000000     1.000000     2.000000   
50%      32.416667     0.500000     2.000000     2.000000     5.000000   
75%      40.066667     1.000000     4.000000     4.000000    10.000000   
max      61.816667     1.000000    38.000000    38.000000    82.000000   

         teamkills   teamdeaths           fb     fbassist  fbvictim  \
count  1080.000000  1080.000000  1080.000000  1080.000000    1080.0   
mean      9.811111     9.822222     0.166667     0.218519       0.0   
std       5.978450     5.990869     0.372851     0.413433       0.0   
min       0.000000     0.000000     0.000000     0.000000       0.0   
25%       5.000000     5.000000     0.000000     0.000000       0.0   
50%      10.000000    10.000000     0.000000     0.000000       0.0   
75%      13.000000    13.000000     0.000000     0.000000       0.0   
max      38.000000    38.000000     1.000000     1.000000       0.0   

            fbtime          kpm         okpm         ckpm           fd  \
count  1080.000000  1080.000000  1080.000000  1080.000000  1080.000000   
mean      1.461110     0.030954     0.029169     0.129212     0.059259   
std       3.418253     0.099132     0.103364     0.272978     0.236219   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max      17.685533     0.750000     1.620469     2.004264     1.000000   

            fdtime  teamdragkills  oppdragkills   elementals  oppelementals  \
count  1077.000000    1080.000000   1080.000000  1080.000000    1080.000000   
mean      2.709421       0.401852      0.319444     0.187963       0.141667   
std       5.345495       0.947949      0.833048     0.615556       0.494010   
min       0.000000       0.000000      0.000000     0.000000       0.000000   
25%       0.000000       0.000000      0.000000     0.000000       0.000000   
50%       0.000000       0.000000      0.000000     0.000000       0.000000   
75%       0.000000       0.000000      0.000000     0.000000       0.000000   
max      22.069133       5.000000      5.000000     4.000000       4.000000   

        firedrakes  waterdrakes  earthdrakes    airdrakes       elders  \
count  1080.000000  1080.000000  1080.000000  1080.000000  1080.000000   
mean      0.058333     0.037037     0.038889     0.053704     0.017593   
std       0.284484     0.203123     0.244242     0.263443     0.157205   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       3.000000     2.000000     3.000

# Understanding the Data from just a Describe
## Basic information we can gather
### Patch 7.18
* 1080 entries most are repeated due to counting each player
* gamelength id about 34.81 minutes
* result are 1, 0 either 1 win or 0 lose
* k =  total kills 3.27 avg max was 38
* d = total deaths 3.27 avg deaths max 38 
kills and deaths seem to correlated
* a = total assits 8.51 avg max 82
* teamkills = 9.811 avg team deaths 9.82
* fb = firstblood occurences avg .21
* fbvictim = 0 in every count
* fb time -  avg 1.46
* kpm kills pe minute
* opponetn kills per miunute
* combined kills per minute ckpm (team kills + opponent kills)
* fd firstdragon rate
* fd time
* teamdragkills
* oppdragkills
* types of dragons
* ft = firsttower
* wcpm wards cleared per min


In [35]:
#remove columns like patch number, url, week, split, league
play_in_df.columns


Index([u'gameid', u'url', u'league', u'split', u'date', u'week', u'game',
       u'patchno', u'playerid', u'side', u'position', u'player', u'team',
       u'champion', u'ban1', u'ban2', u'ban3', u'ban4', u'ban5', u'gamelength',
       u'result', u'k', u'd', u'a', u'teamkills', u'teamdeaths', u'doubles',
       u'triples', u'quadras', u'pentas', u'fb', u'fbassist', u'fbvictim',
       u'fbtime', u'kpm', u'okpm', u'ckpm', u'fd', u'fdtime', u'teamdragkills',
       u'oppdragkills', u'elementals', u'oppelementals', u'firedrakes',
       u'waterdrakes', u'earthdrakes', u'airdrakes', u'elders', u'oppelders',
       u'herald', u'heraldtime', u'ft', u'fttime', u'firstmidouter',
       u'firsttothreetowers', u'teamtowerkills', u'opptowerkills', u'fbaron',
       u'fbarontime', u'teambaronkills', u'oppbaronkills', u'dmgtochamps',
       u'dmgtochampsperminute', u'dmgshare', u'earnedgoldshare', u'wards',
       u'wpm', u'wardshare', u'wardkills', u'wcpm', u'visionwards',
       u'visionwardbuys',

In [66]:
df = play_in_df.drop(['url','league','split','week','patchno'],axis=1)
df.head(15)

gameid          date game  playerid  side position      player  \
0   1002360213  43001.001840    1         1  Blue      Top         957   
1   1002360213  43001.001840    1         2  Blue   Jungle       Condi   
2   1002360213  43001.001840    1         3  Blue   Middle        xiye   
3   1002360213  43001.001840    1         4  Blue      ADC      Mystic   
4   1002360213  43001.001840    1         5  Blue  Support         Ben   
5   1002360213  43001.001840    1         6   Red      Top      Jirall   
6   1002360213  43001.001840    1         7   Red   Jungle       Oddie   
7   1002360213  43001.001840    1         8   Red   Middle       Seiya   
8   1002360213  43001.001840    1         9   Red      ADC  WhiteLotus   
9   1002360213  43001.001840    1        10   Red  Support     Genthix   
10  1002360213  43001.001840    1       100  Blue     Team        Team   
11  1002360213  43001.001840    1       200   Red     Team        Team   
12  1002360222  43001.060671    2         1  Blue      Top      Impact   
13  1002360222  43001.060671    2         2  Blue   Jungle   Contractz   
14  1002360222  43001.060671    2         3  Blue   Middle      Jensen   

           team  champion      ban1     ban2       ban3     ban4       ban5  \
0       Team WE  Cho'gath   Rek'Sai  Alistar   Tristana    Karma       Shen   
1       Team WE    Gragas   Rek'Sai  Alistar   Tristana    Karma       Shen   
2       Team WE     Corki   Rek'Sai  Alistar   Tristana    Karma       Shen   
3       Team WE     Xayah   Rek'Sai  Alistar   Tristana    Karma       Shen   
4       Team WE     Janna   Rek'Sai  Alistar   Tristana    Karma       Shen   
5   Lyon Gaming    Maokai   Kalista  Sejuani  Jarvan IV  LeBlanc      Galio   
6   Lyon Gaming   Lee Sin   Kalista  Sejuani  Jarvan IV  LeBlanc      Galio   
7   Lyon Gaming    Syndra   Kalista  Sejuani  Jarvan IV  LeBlanc      Galio   
8   Lyon Gaming   Kog'Maw   Kalista  Sejuani  Jarvan IV  LeBlanc      Galio   
9   Lyon Gaming     Braum   Kalista  Sejuani  Jarvan IV  LeBlanc      Galio   
10      Team WE             Rek'Sai  Alistar   Tristana    Karma       Shen   
11  Lyon Gaming             Kalista  Sejuani  Jarvan IV  LeBlanc      Galio   
12       Cloud9     Galio  Tristana  Sejuani    Kalista  Trundle  Dr. Mundo   
13       Cloud9   Rek'Sai  Tristana  Sejuani    Kalista  Trundle  Dr. Mundo   
14       Cloud9   LeBlanc  Tristana  Sejuani    Kalista  Trundle  Dr. Mundo   

    gamelength  result   k   d   a  teamkills  teamdeaths doubles triples  \
0    47.166667       1   2   4   5         11          15       0       0   
1    47.166667       1   0   4   8         11          15       0       0   
2    47.166667       1   6   1   5         11          15       0       0   
3    47.166667       1   3   3   5         11          15       0       0   
4    47.166667       1   0   3  10         11          15       0       0   
5    47.166667       0   0   3  13         15          11       0       0   
6    47.166667       0   2   3  10         15          11       0       0   
7    47.166667       0   5   1   9         15          11       1       0   
8    47.166667       0   8   1   7         15          11       2       0   
9    47.166667       0   0   3  15         15          11       0       0   
10   47.166667       1  11  15  33         11          15                   
11   47.166667       0  15  11  54         15          11                   
12   30.250000       1   2   0   6         13           3       0       0   
13   30.250000       1   2   2   5         13           3       0       0   
14   30.250000       1   5   1   7         13           3       0       0   

   quadras pentas  fb  fbassist  fbvictim     fbtime       kpm      okpm  \
0        0      0   0         0         0   0.000000  0.000000  0.000000   
1        0      0   0         0         0   0.000000  0.000000  0.000000   
2        0      0   0         0         0   0.000000  0.000000  0.000000   
3        0      0   0

In [67]:
#seperate teams from players


In [68]:
df_team = df.loc[df['position'] == 'Team']
df_team.head(20)
df_team.reset_index(inplace=True)
#remove champion, doubles, triples, quadras, pentas


In [69]:
df_team.drop(['index','champion','doubles','triples','quadras','pentas','playerid'], axis=1, inplace=True)
df_team.head(10)

C:\ProgramData\Anaconda2\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


gameid          date game  side position player              team  \
0  1002360213  43001.001840    1  Blue     Team   Team           Team WE   
1  1002360213  43001.001840    1   Red     Team   Team       Lyon Gaming   
2  1002360222  43001.060671    2  Blue     Team   Team            Cloud9   
3  1002360222  43001.060671    2   Red     Team   Team  Team oNe Esports   
4  1002360229  43001.101134    3  Blue     Team   Team    Gambit Esports   
5  1002360229  43001.101134    3   Red     Team   Team       Lyon Gaming   
6  1002360235  43001.141991    4  Blue     Team   Team       Dire Wolves   
7  1002360235  43001.141991    4   Red     Team   Team            Cloud9   
8  1002360244  43001.178218    5  Blue     Team   Team    Gambit Esports   
9  1002360244  43001.178218    5   Red     Team   Team           Team WE   

       ban1        ban2       ban3      ban4        ban5  gamelength  result  \
0   Rek'Sai     Alistar   Tristana     Karma        Shen   47.166667       1   
1   Kalista     Sejuani  Jarvan IV   LeBlanc       Galio   47.166667       0   
2  Tristana     Sejuani    Kalista   Trundle   Dr. Mundo   30.250000       1   
3     Elise      Thresh     Syndra   Orianna   Jarvan IV   30.250000       0   
4    Lucian       Galio    Alistar    Syndra     LeBlanc   33.383333       0   
5   Sejuani  Cassiopeia    Kalista      Kayn      Anivia   33.383333       1   
6    Maokai        Shen    Orianna    Thresh  Blitzcrank   27.850000       0   
7   Kalista       Rakan      Jayce  Tristana     Kog'Maw   27.850000       1   
8     Galio       Xayah    Alistar   LeBlanc      Lucian   29.350000       0   
9   Sejuani   Jarvan IV    Kalista     Brand      Thresh   29.350000       1   

    k   d   a  teamkills  teamdeaths  fb  fbassist  fbvictim     fbtime  \
0  11  15  33         11          15   0         0         0   0.000000   
1  15  11  54         15          11   1         1         0   9.984617   
2  13   3  33         13           3   1         1         0  11.917667   
3   3  13   5          3          13   0         0         0   0.000000   
4   2  14   6          2          14   0         0         0   0.000000   
5  14   2  34         14           2   1         1         0   6.104567   
6   3  11   6          3          11   0         0         0   0.000000   
7  11   3  24         11           3   1         1         0   4.418217   
8   8  24  14          8          24   1         1         0   2.722633   
9  24   8  55         24           8   0         0         0   0.000000   

        kpm      okpm      ckpm  fd     fdtime  teamdragkills  oppdragkills  \
0  0.000000  0.000000  0.000000   0   0.000000              0             0   
1  0.318021  0.233216  0.551237   0   9.857433              2             3   
2  0.429752  0.099174  0.528926   1  11.573750              3             0   
3  0.000000  0.000000  0.000000   0   0.000000              0             0   
4  0.000000  0.000000  0.000000   0   0.000000              0             0   
5  0.419371  0.059910  0.479281   1  12.224950              3             0   
6  0.000000  0.000000  0.000000   0   0.000000              0             0   
7  0.394973  0.107720  0.502693   1   8.451467              3             0   
8  0.272572  0.817717  1.090290   1   9.634017              2             1   
9  0.000000  0.000000  0.000000   0   0.000000              0             0   

   elementals  oppelementals  firedrakes  waterdrakes  earthdrakes  airdrakes  \
0           0              0           0            0            0          0   
1           2              2           1            0            0          1   
2           3              0           0            1            2          0   
3           0              0           0            0            0          0   
4           0              0           0            0            0          0   
5           3              0           2            0            0          1   
6           0             

In [71]:
df_team.head(100)

gameid          date game  side position player                 team  \
0   1002360213  43001.001840    1  Blue     Team   Team              Team WE   
1   1002360213  43001.001840    1   Red     Team   Team          Lyon Gaming   
2   1002360222  43001.060671    2  Blue     Team   Team               Cloud9   
3   1002360222  43001.060671    2   Red     Team   Team     Team oNe Esports   
4   1002360229  43001.101134    3  Blue     Team   Team       Gambit Esports   
5   1002360229  43001.101134    3   Red     Team   Team          Lyon Gaming   
6   1002360235  43001.141991    4  Blue     Team   Team          Dire Wolves   
7   1002360235  43001.141991    4   Red     Team   Team               Cloud9   
8   1002360244  43001.178218    5  Blue     Team   Team       Gambit Esports   
9   1002360244  43001.178218    5   Red     Team   Team              Team WE   
10  1002360251  43001.219190    6  Blue     Team   Team          Dire Wolves   
11  1002360251  43001.219190    6   Red     Team   Team     Team oNe Esports   
12  1002360287  43001.989317    1  Blue     Team   Team     Team oNe Esports   
13  1002360287  43001.989317    1   Red     Team   Team               Cloud9   
14  1002360296  43002.032998    2  Blue     Team   Team          Lyon Gaming   
15  1002360296  43002.032998    2   Red     Team   Team              Team WE   
16  1002360302  43002.076539    3  Blue     Team   Team               Cloud9   
17  1002360302  43002.076539    3   Red     Team   Team          Dire Wolves   
18  1002360310  43002.110116    4  Blue     Team   Team          Lyon Gaming   
19  1002360310  43002.110116    4   Red     Team   Team       Gambit Esports   
20  1002360317  43002.152674    5  Blue     Team   Team     Team oNe Esports   
21  1002360317  43002.152674    5   Red     Team   Team          Dire Wolves   
22  1002360324  43002.191944    6  Blue     Team   Team              Team WE   
23  1002360324  43002.191944    6   Red     Team   Team       Gambit Esports   
24  1002360332  43002.224711    T  Blue     Team   Team          Dire Wolves   
25  1002360332  43002.224711    T   Red     Team   Team     Team oNe Esports   
26  1002360365  43002.989062    1  Blue     Team   Team     Young Generation   
27  1002360365  43002.989062    1   Red     Team   Team    Kaos Latin Gamers   
28  1002360372  43003.052350    2  Blue     Team   Team              Rampage   
29  1002360372  43003.052350    2   Red     Team   Team   Hong Kong Attitude   
30  1002360382  43003.088113    3  Blue     Team   Team               Fnatic   
31  1002360382  43003.088113    3   Red     Team   Team    Kaos Latin Gamers   
32  1002360389  43003.121331    4  Blue     Team   Team   Hong Kong Attitude   
33  1002360389  43003.121331    4   Red     Team   Team   Fenerbahce Esports   
34  1002360398  43003.186250    5  Blue     Team   Team     Young Generation   
35  1002360398  43003.186250    5   Red     Team   Team               Fnatic   
36  1002360404  43003.262326    6  Blue     Team   Team              Rampage   
37  1002360404  43003.262326    6   Red     Team   Team   Fenerbahce Esports   
38  1002360429  43003.989375    1  Blue     Team   Team   Hong Kong Attitude   
39  1002360429  43003.989375    1   Red     Team   Team              Rampage   
40  1002360441  43004.033819    2  Blue     Team   Team    Kaos Latin Gamers   
41  1002360441  43004.033819    2   Red     Team   Team     Young Generation   
42  1002360452  43004.070093    3  Blue     Team   Team   Fenerbahce Esports   
43  1002360452  43004.070093    3   Red     Team   Team              Rampage   
44  1002360463  43004.111146    4  Blue     Team   Team    Kaos Latin Gamers   
45  1002360463  43004.111146    4   Red     Team   Team               Fnatic   
46  1002360472  43004.146111    5  Blue     Team   Team   Fenerbahce Esports   
47  1002360472  43004.146111    5   Red     Team   Team   Hong Kong Attitude   
48  1002360478  43004.192546    6  Blue     Team   Team               Fnatic   
49  1002